convert_image.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/bash
+
+###########################################################
+#
+# Author: noirmurir
+# Sending_Grounds Bash Scripting Challenge
+#
+# Shell frontend that processes the filenames and calls
+# the correct conversion utilities.
+# Optionally we can change the size of the resulting image,
+# place a colored border around an image
+# NetPBM utilities are used in this script to 
+# work with the images.
+#
+# Requirements:
+# - Catches signals
+# - Has a help/usage statement
+# - Uses named Exit codes
+# - Has arg parsing
+# - Posts it to this repo here
+# - and the script of course works
+# 
+###########################################################
+
+# Named exit codes
+EXIT_SUCCESS=0
+EXIT_INVALID_ARGS=1
+EXIT_SIGNAL_RECEIVED=2
+
+# Print help/usage
+usage() {
+cat<<EOF
+Usage: $0 [-S] [-s N] [-w N] [-c S] imagefile...
+
+Options:
+  -S          Change sharpness of a resulting image
+  -s N        Change size of a resulting image
+  -w N        Change width of a border of a resulting image
+  -c S        Change color of a border of a resulting image
+  -h          Show help/usage
+EOF
+}
+
+handle_signals() {
+    echo "Recieved signal - Exiting gracefully..."
+    exit $EXIT_SIGNAL_RECEIVED
+}
+
+# Main function serving as entrypoint
+main() {
+    # Trap signals
+    trap 'handle_signals' SIGINT SIGTERM
+
+    # Set up default width and color
+    width=1
+    colour='-color grey'
+
+    # Initialise the pipeline components
+    standardise=' | pnmtojpeg -quiet'
```

Comment:
One thing you might consider is having some form of a `setup()` func where you can do some of the initial setup, or in this case you're using `pnmnlfilt` and associated utilities but we don't have a check to see if they are installed or in our PATH. Bit of a nicety, but it makes it a better user experience.

Basically just to tackle the below for them/yourself.

```bash
[Sending_Grounds@HomeBase ~]$ type pnmnlfilt
-bash: type: pnmnlfilt: not found
[Sending_Grounds@HomeBase ~]$ echo $?
1
[Sending_Grounds@HomeBase ~]$ sudo yum whatprovides pnmnlfilt
...<TRUNCATED>...
netpbm-progs-10.79.00-7.amzn2.x86_64 : Tools for manipulating graphics files in netpbm supported formats
Repo        : amzn2-core
Matched from:
Filename    : /usr/bin/pnmnlfilt

[Sending_Grounds@HomeBase ~]$ sudo yum install netpbm-progs-10.79.00-7.amzn2.x86_64 -y
Loaded plugins: extras_suggestions, langpacks, priorities, update-motd
Resolving Dependencies
--> Running transaction check
...<TRUNCATED>...
Complete!
[Sending_Grounds@HomeBase ~]$ type pnmnlfilt
pnmnlfilt is /usr/bin/pnmnlfilt
[Sending_Grounds@HomeBase ~]$ echo $?
0
```